### PR TITLE
Improve error reporting in examples and documentation

### PR DESCRIPTION
**Pull request by clue**
Refs https://github.com/reactphp/http/pull/427, https://github.com/reactphp/promise-stream/pull/26 and others

### CODE REVIEW

File: examples/archive.php [hunk 1]
```diff
@@ -27,13 +27,14 @@
     });
 });
 
-$tar->on('error', function ($e = null) {
+$tar->on('error', function (Exception $e) {
     // should not be invoked, unless the stream is somehow interrupted
-    echo 'ERROR processing tar stream' . PHP_EOL . $e;
+    echo 'ERROR processing tar stream' . $e->getMessage() . PHP_EOL;
```
Comment (SimonFrings) [inline examples/archive.php:32]: Nit: Maybe add a colon in 32 as well? (reference line 37)

File: examples/archive.php [hunk 2]
```diff
@@ -27,13 +27,14 @@
...
 });
-$stream->on('error', function ($e = null) {
+
+$stream->on('error', function (Exception $e) {
```
Comment (SimonFrings) [inline examples/archive.php:35]: In other examples you changed the same error message to `echo 'Error: ' . $e->getMessage() . PHP_EOL;`.
Any reason why this one is different?

File: examples/archive.php [hunk 3]
```diff
@@ -27,13 +27,14 @@
...
     // will be called if either parameter is invalid
-    echo 'ERROR requesting stream' . PHP_EOL . $e;
+    echo 'ERROR requesting stream: ' . $e->getMessage() . PHP_EOL;
```
Comment (SimonFrings) [inline examples/archive.php:37]: In other examples you changed the same error message to `echo 'Error: ' . $e->getMessage() . PHP_EOL;`.
Any reason why this one is different?

**Review by SimonFrings — CHANGES_REQUESTED** [review]

File: examples/archive.php (hunk 1 above)
Comment (clue) [reply examples/archive.php]: Good catch, updated as suggested :+1:

File: examples/archive.php (hunk 3 above)
Comment (clue) [reply examples/archive.php]: Good catch, updated as suggested :+1:

**Pull request merged**